ruby/ql/lib/codeql/ruby/frameworks/ActiveStorage.qll

```diff
@@ -8,23 +8,236 @@ private import codeql.ruby.Concepts
+            .getAMethodCall("sanitized")
+    }
   }
-}
 
-/** Taint related to `ActiveStorage::Filename`. */
-private class Summaries extends ModelInput::SummaryModelCsv {
-  override predicate row(string row) {
-    row =
-      [
-        "activestorage;;Member[ActiveStorage].Member[Filename].Method[new];Argument[0];ReturnValue;taint",
-        "activestorage;;Member[ActiveStorage].Member[Filename].Instance.Method[sanitized];Argument[self];ReturnValue;taint",
-      ]
+  /** Taint related to `ActiveStorage::Filename`. */
+  private class FilenameSummaries extends ModelInput::SummaryModelCsv {
+    override predicate row(string row) {
+      row =
+        [
+          "activestorage;;Member[ActiveStorage].Member[Filename].Method[new];Argument[0];ReturnValue;taint",
+          "activestorage;;Member[ActiveStorage].Member[Filename].Instance.Method[sanitized];Argument[self];ReturnValue;taint",
+        ]
+    }
+  }
+
+  /**
+   * `Blob` is an instance of `ActiveStorage::Blob`.
+   */
+  private class BlobTypeSummary extends ModelInput::TypeModelCsv {
+    override predicate row(string row) {
+      // package1;type1;package2;type2;path
+      row =
+        [
+          // ActiveStorage::Blob.new : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Instance",
+          // ActiveStorage::Blob.create_and_upload! : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[create_and_upload!].ReturnValue",
+          // ActiveStorage::Blob.create_before_direct_upload! : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[create_before_direct_upload!].ReturnValue",
+          // ActiveStorage::Blob.compose(blobs : [Blob]) : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[compose].ReturnValue",
+          // gives error: Invalid name 'Element' in access path
+          // "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[compose].Argument[0].Element[any]",
+          // ActiveStorage::Blob.find_signed(!) : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[find_signed,find_signed!].ReturnValue",
+          // ActiveStorage::Attachment#blob : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Attachment].Instance.Method[blob].ReturnValue",
+          // ActiveStorage::Attachment delegates method calls to its associated Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Attachment].Instance",
+        ]
+    }
+  }
+
+  /**
+   * Method calls on `ActiveStorage::Blob` that send HTTP requests.
+   */
+  private class BlobRequestCall extends HTTP::Client::Request::Range {
+    BlobRequestCall() {
+      this =
+        [
+          // Class methods
+          API::getTopLevelMember("ActiveStorage")
+              .getMember("Blob")
+              .getASubclass*()
```

Comment:
Is the transitive closure necessary here? `getASubclass()` is defined as `Node getASubclass() { result = this.getAnImmediateSubclass*() }`